api-controllers/ServiceFabrikApiController.js

```diff
@@ -427,16 +434,27 @@ class ServiceFabrikApiController extends FabrikBaseController {
           backup_guid: backupGuid,
           tenant_id: tenantId
         };
-        return this.backupStore
-          .getBackupFile(backupFileOptions);
+        return timeStamp ?
+          this.backupStore
+          .listBackupsOlderThan(backupFileOptions, new Date(timeStamp))
+          .then(oldBackupArray =>
+            _.sortBy(oldBackupArray, ['started_at']))
+          .then(sortedArray =>
+            _.findLast(sortedArray, backup => backup.state ===  CONST.OPERATION.SUCCEEDED))
+          .tap(successfulBackup => {
+            if (_.isEmpty(successfulBackup)) {
+              throw new NotFound(`No backup successful found for service instance '${instanceId}' before time_stamp ${new Date(timeStamp)} `);
+            }
+          }) :
+          this.backupStore.getBackupFile(backupFileOptions);
       })
-      .catchThrow(NotFound, new UnprocessableEntity(`No backup with guid '${backupGuid}' found in this space`))
+      .catchThrow(NotFound, new UnprocessableEntity(`Cannot restore for guid/timeStamp '${timeStamp || backupGuid}' found in this space`))
       .tap(metadata => {
         if (metadata.state !== 'succeeded') {
```

Comment:
In case of timestamp based restore, from [this block](https://github.com/sauravmndl/service-fabrik-broker/blob/a5e632e5af51476105095a147e39cef86b0395d0/api-controllers/ServiceFabrikApiController.js#L437-L448), where will we get metadata from?

Reply:
.listBackupsOlderThan(backupFileOptions, new Date(timeStamp)) provides metadata.